src/Serializer/SnippetDenormalizer.php

```diff
@@ -30,16 +30,16 @@ public function denormalizeSnippet(array $item) : PromiseInterface
         $this->identityMap->reset($id);
 
         if (empty($this->globalCallback)) {
-            $this->globalCallback = new CallbackPromise(function () {
+            $this->globalCallback = new Promise(function () {
                 $this->identityMap->fillMissingWith($this->fetchComplete);
 
-                $this->globalCallback = null;
-
                 $settled = $this->identityMap->waitForAll();
 
                 $this->identityMap->reset();
 
-                return $settled;
+                $this->globalCallback->resolve($settled);
```

Comment:
Yeah, they're delayed and batched. We need this delayed requesting to stop Guzzle from queuing a whole bunch of requests that are probably not needed (requesting something else straight after will cause them to send). The batching helps these requests be sent in parallel (it assume that if you want one, you also want the others).